### DISCUSSION TITLE
API rule only requiring user is authenticated

### DISCUSSION THREAD

**Discussion by JGStJohnson** [pocketbase/pocketbase#5948] [Q&A] [answered]
Hi, I'm looking through the documentation [here](https://pocketbase.io/docs/api-rules-and-filters/) and through the other discussions, and found #1157 which explained how it's done but I'm still a little confused.

For the create/read/update/delete I want to enforce that they are authenticated first.
In #1157 the example given was
`// create ("user = @request.auth.id" will also work for create)
@request.data.user = `@request.auth.id`

Could someone please tell me how this would work?
It's also mentioned that the user is considered authenticated if they pass a valid Authorization:TOKEN header.

My first thought would be to do `@request.headers.Authorization != ''` but even if it is not empty, I can't validate it.

Is it safe to say that if the request has `@request.auth.*` that the user is authenticated?

**Comment by JGStJohnson** [comment]
In other words how can I deny access to anonymous users.

**Reply by ganigeorgiev** [reply, marked as answer]
Please try to be more specific. Your questions are too vague and I don't understand what do you mean by _"how this would work"_.


In any case to deny access to anonymous users you can use a rule like `@request.auth.id != ""`.
If the user is authenticated `@request.auth.id` is always populated.

_Also note that with PocketBase v0.23+ `@request.data.*` is renamed to `@request.body.*`._

**Reply by JGStJohnson** [reply]
My apologies, when asking "how this would work" I was asking about `@request.body.user = @request.auth.id` specifically.
I haven't dove too deep into creating the rules for my application yet, so I haven't been able to see what fields are available. I did read that you have implemented autocomplete to help with that task so that will be able to help me.

As for the question of how to deny unauthenticated users, you have answered that and I will close the discussion.

Thank you for your help.
